File: go1/basico_intermediario_interativo.go
package main

import "fmt"

func main() {

	// Normal

	// Nome
	
	var nome string
	fmt.Print("Digite seu nome:")
	fmt.Scanln(&nome) // Equivalente ao input do python
	fmt.Println("Olá,", nome)

	// Idade

    var idade int
	fmt.Print("Digite sua idade:")
	fmt.Scanln(&idade)
	fmt.Println("Sua idade é:", idade)

	// Tratamento de erros

    // Nome

	var nome1 string
	fmt.Println("Digite seu nome:")

	if _, err := fmt.Scanln(&nome1); err != nil {
		fmt.Println("Erro na entrada:", err)
		return
	}
    
	fmt.Println("Seu nome é:", nome1)

	// Idade

	var idade0 int
	fmt.Println("Digite sua idade:")

	if _, err := fmt.Scanln(&idade0); err != nil {
		fmt.Println("Erro na entrada:", err)
		return
	}

    fmt.Println("Sua idade é:", idade0)

	// While true (for infinito) com tratamento de erros incluído

	var texto string
    fmt.Println("Digite um texto (ou 'sair' para sair):")

    for {
        fmt.Print("Saída:")
        _, err := fmt.Scanln(&texto)

        if err != nil {
            fmt.Println("Erro na digitação do texto:", err)
            continue // continue ou break
        }

        if texto == "sair" {
            fmt.Println("Encerrado.")
            break
        }

        fmt.Println("Texto:", texto)
    }
}

File: py/basico_intermediario_interativo.py
nome = input("Digite seu nome:")
print(f"Seu nome é {nome}!")

# tentativa de cliente com tratamento de erros simples e sem controle de fluxo
try:
    idade = int(input("Digite sua idade:"))
    if idade <= 0:
        raise ValueError("Idade nao pode ser abaixo de zero!")
    print(f"A sua idade é {idade}")
except ValueError:
    print("Erro: Entrada inválida!")
finally:
    print("Fim da execução.")

# imput com condicionais e com controle de fluxo
nome = input("Digite seu nome")
if nome in ["Jose", "Roberto", "Hermon"]:
    print("Acesso aceito")
else:
    print("Acesso negado")

for i in range(2):
    nome = input(f"Digite seu nome {i + 1}: ")

# funçao com input


def saudacoes():
    nome = input("Digite seu nome:")
    print(f"Seja bem-vindo, {nome} !")


saudacoes()
